presto-main/src/main/java/io/prestosql/cost/EnforceSingleRowStatsRule.java

```diff
@@ -42,8 +47,10 @@ public EnforceSingleRowStatsRule(StatsNormalizer normalizer)
     @Override
     protected Optional<PlanNodeStatsEstimate> doCalculate(EnforceSingleRowNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
     {
-        return Optional.of(PlanNodeStatsEstimate.buildFrom(sourceStats.getStats(node.getSource()))
-                .setOutputRowCount(1)
-                .build());
+        Map<Symbol, SymbolStatsEstimate> symbolStatsEstimate = sourceStats.getStats(node.getSource()).getSymbolStatistics()
+                .entrySet()
+                .stream()
+                .collect(Collectors.toMap(Entry::getKey, entry -> entry.getValue().mapDistinctValuesCount(distinctValueCount -> min(distinctValueCount, 1))));
```

Comment:
`StatsNormalizer` should do this. Isn't it doing this?

Reply:
Ah, I missed the normalization part. I found the the inconsistency within a rule while analyzing a query. Tested, this works for stats case.
For no column stats case (but row count is present), is it reasonable to add column stats with NDV as 1 for EnforceSingleRowNode?

Reply:
> For no column stats case (but row count is present), is it reasonable to add column stats with NDV as 1 for EnforceSingleRowNode?

no; NDV doesn't include null value

Reply:
@rohangarg are you working on this PR?

Reply:
@sopel39 : No, I'm not working on this. 
As Piotr mentioned, approximating 1 as NDV for `EnforceSingleRowStatsRule` can be incorrect - so it would be better to keep it as is.

Reply:
> so it would be better to keep it as is.

Ok, So I think we should close this PR then